### PR TITLE
Uptake oc sdk

### PR DESCRIPTION
**Pull request by mithunjohny**
# PR Details

### Description

_Include a description of the problem to be solved_

## Solution Proposed

This pull request includes several updates to improve error handling, enhance robustness, and update dependencies in the `OmnichannelChatSDK`. The most important changes focus on refining exception handling, ensuring fallback values for optional parameters, and updating a dependency version.

### Error Handling Improvements:
* Enhanced the `throwChatSDKError` function in `src/utils/exceptionThrowers.ts` to include detailed error information when the error object contains a `response` property. This ensures better debugging and logging capabilities by serializing relevant fields like `status`, `headers`, and `data`.

### Robustness Enhancements:
* Updated the `closeChat` method in `src/OmnichannelChatSDK.ts` to ensure that the `disconnect` logic is executed in a `finally` block, guaranteeing cleanup actions even if an error occurs during `closeChat`. [[1]](diffhunk://#diff-195caf6fab2fa65f9272d22a54b5688ff7f481b748ba940a93ed1e46df65a550R937-R939) [[2]](diffhunk://#diff-195caf6fab2fa65f9272d22a54b5688ff7f481b748ba940a93ed1e46df65a550R965)
* Modified the handling of `optionalParams.liveChatContext` in `src/OmnichannelChatSDK.ts` to use fallback values (`this.requestId` and `this.chatToken`) when `requestId` or `chatToken` is not provided. This prevents potential null or undefined errors.

### Dependency Updates:
* Updated the `@microsoft/ocsdk` dependency version in `package.json` from `^0.5.14` to `^0.5.15`, likely to incorporate bug fixes or new features.

### Acceptance criteria

_Define what are the conditions to consider the PR has achieved the intended goal_

## Test cases and evidence

_Include what tests cases were considered, any evidence of testing for future references, to identify any corner cases, etc_

### Sanity Tests

- [ ] You have tested all changes in Popout mode
- [x] You have tested all changes in cross browsers i.e Edge, Chrome, Firefox, Safari and mobile devices(iOS and Android)
- [x] Your changes are included in the [CHANGELOG](../CHANGE_LOG.md)

## A11y

- [ ] You have run the [Automated Accessibility Check](https://accessibilityinsights.io/docs/en/windows/getstarted/automatedchecks) and have no reported issues

***Please provide justification if any of the validations has been skipped.***

### CODE REVIEW

**Review by elopezanaya — APPROVED** [review]

**Review by Sonam-Ligade — APPROVED** [review]

File: src/OmnichannelChatSDK.ts [hunk 1]
```diff
@@ -1021,8 +1024,8 @@ class OmnichannelChatSDK {
         let sessionId = this.sessionId;
 
         if (optionalParams.liveChatContext) {
-            requestId = optionalParams.liveChatContext.requestId;
-            chatToken = optionalParams.liveChatContext.chatToken;
+            requestId = optionalParams.liveChatContext.requestId || this.requestId;
```
Comment (xTEddie) [inline src/OmnichannelChatSDK.ts:1027]: Why should `this.requestId` and `this.chatToken` be passed?

ChatSDK should not passed any of these because this should be something a `costumer` explicitly passes. It's the "chat" a customer decides to connect back to. We cannot alter it even if it's invalid. Otherwise we could potentially alter the intended behavior of the customer.

We can update the `if` statement to ensure `optionalParams.liveChatContext.requestId` and `optionalParams.liveChatContext.chatToken` are present to validate it's a valid `liveChatContext` format

**Review by xTEddie — APPROVED** [review]
